lib/index.js

```diff
@@ -0,0 +1,106 @@
+import { SESv2 } from "@aws-sdk/client-sesv2";
+import { ProductHistory, Status, User } from "@podaac/swodlr-common/lib/models.js";
+import { sdsStatuses } from "@podaac/swodlr-common";
+import { Op } from "sequelize";
+import utils from "./utilities.js";
+
+const logger = utils.getLogger(import.meta.url);
+const productReadyTemplate = await utils.loadTemplate('product-ready');
+const validateJob = utils.loadSchema('job');
+
+const ses = new SESv2({
+  credentials: JSON.parse(utils.getParameter('ses_credentials')),
+  region: utils.getParameter('ses_region')
+});
+
+export async function lambdaHandler(event) {
+  logger.debug(`Records received: ${event['Records'].length}`);
+
+  let jobs = [];
+  for (const record of event['Records']) {
+    const job = JSON.parse(record['body']);
+    if (validateJob(job)) {
+      jobs = jobs.concat([job]);
+    } else {
+      logger.error('Job failed to validate');
+      logger.debug(`Failed job: ${JSON.stringify(job)}`);
+    }
+  }
+  logger.debug(`Jobs received: ${jobs.length}`);
+
+  for (const job of jobs) {
+    logger.info(`Processing job: ${job['job_id']}`);
+
+    const jobStatus = job['job_status'];
+    if (!sdsStatuses.SUCCESS.has(jobStatus)) {
+      continue;
+    }
+
+    const stage = job['stage'];
+    if (stage !== 'submit_raster') {
+      continue;
+    }
+
+    const productId = job['product_id'];
+
+    const statusQueryResults = await Status.findAll({
+      where: {
+        state: 'NEW', // TODO: Move this to a const in commons
```

Comment:
Shouldn't we be looking for products that are in the 'AVAILABLE' state? Why are we searching for 'NEW' here?

Reply:
Its because this service is invoked asynchronously along with async-update right after the raster-create pipeline finishes execution so if async-update is behind on updating RDS, we would run into a race condition where user-notify would try to check the database for statuses that are still processing on async-update

Reply:
What if async-update finishes before user-notify?

Couldn't async-update be publishing the status change event from ready->available and user-notify should be listening for 'available' status change events?

Reply:
> What if async-update finishes before user-notify?
We specifically only look for the NEW status so it doesn’t matter which service executes first

> Couldn't raster-create be publishing the status change event from ready->available and user-notify should be listening for 'available' status change events?
The problem is that the change from NEW->READY/AVAILABLE occurs all in one message from raster-create since we also publish the granules immediately after the SDS pipeline finishes. We would need to send out two messages from the point of the SDS pipeline completion: one for right before publish and one right after

Reply:
So the logic is basically if

1. The notification message has job status SUCCESS
2. The notification message is the submit_raster stage
3. The product status is at least NEW

That means a user notification should be sent?

Reply:
Yes, those are the trigger conditions

Reply:
I'm a little concerned that some of the SDS specifics are leaking out of the components that interface with the SDS; was hoping to keep that isolated to only the parts that need to know about it and rely on the swodlr domain objects for the rest of the system. But I understand the logic now so that's good; perhaps we should have another architecture discussion now that we're deeper into implementation.

Reply:
The main issue I see for violating the isolation is that we have “job_status” straight from the SDS; we can abstract those out to SUCCESS/FAIL/WAITING on our end, but we’d need to start that change on raster-create before pushing it here. Probably also update the schemas to state that those are the only three states as well